src/NHibernate/Proxy/DynamicProxy/ProxyFactory.cs

```diff
@@ -151,32 +129,14 @@ private TypeInfo CreateUncachedProxyType(System.Type baseType, System.Type[] bas
 			return proxyType;
 		}
 
-		private IEnumerable<System.Type> GetInterfaces(System.Type currentType)
-		{
-			return GetAllInterfaces(currentType);
-		}
-
-		private IEnumerable<System.Type> GetAllInterfaces(System.Type currentType)
```

Comment:
`System.Type.GetInterfaces()` already returns _all_ implemented interfaces, so no need to further traverse the interfaces tree.